cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -91,51 +201,730 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
   @Override
   public List<String> getScheduleIds(Id.Program program, SchedulableProgramType programType)
     throws SchedulerException {
-    return ImmutableList.of();
+    char startChar = ':';
+    char endChar = (char) (startChar + 1);
+    String programScheduleId = AbstractSchedulerService.programIdFor(program, programType);
+    return ImmutableList.copyOf(scheduleSubscribers.subMap(String.format("%s%c", programScheduleId, startChar),
+                                                           String.format("%s%c", programScheduleId, endChar))
+                                  .keySet());
   }
 
   @Override
   public void suspendSchedule(Id.Program program, SchedulableProgramType programType, String scheduleName)
-    throws NotFoundException, SchedulerException {
-
+    throws ScheduleNotFoundException, SchedulerException {
+    String scheduleId = AbstractSchedulerService.scheduleIdFor(program, programType, scheduleName);
+    StreamSubscriber subscriber = scheduleSubscribers.get(scheduleId);
+    if (subscriber == null) {
+      throw new ScheduleNotFoundException(scheduleName);
```

Comment:
what happens if you schedule a program, but createScheduleTask() fails (see line 178)? schedule() will return without error, so the schedule should exist. But here you throw a ScheduleNotFound, which seems wrong. Either schedule should error out, or this should throw something else?